src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,118 @@
+
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.EventCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.TodoCommand;
+
+/**
+ * This class represents a {@code Parser}. It parses user inputs.
+ *
+ * @author Elizabeth Chow
+ */
+public class Parser {
+    /**
+     * Returns a {@code Command} that contains parsed and trimmed inputs.
+     *
+     * @param input Raw user input.
+     *
+     * @return A {@code Command} containing parsed inputs.
+     * @throws DukeException User input not in the list of predefined keywords.
+     */
+    public static Command parse(String input) throws DukeException {
+        String inputCommand = input.split(" ")[0];
+        String parsedMessage = input.replace(inputCommand, "").trim();
+
+        switch (inputCommand) {
+        case "bye":
+            return parseByeInput();
+        case "deadline":
+            return parseDeadlineInput(parsedMessage);
+        case "delete":
+            return parseDeleteInput(parsedMessage);
+        case "done":
+            return parseDoneInput(parsedMessage);
+        case "event":
+            return parseEventInput(parsedMessage);
+        case "find":
+            return parseFindInput(parsedMessage);
+        case "list":
+            return parseListInput();
+        case "todo":
+            return parseTodoInput(parsedMessage);
+        default:
+            throw new DukeException("I'm sorry, but I don't know what that means :-(");
+        }
+    }
+
+    private static Command parseByeInput() {
+        return new ExitCommand();
+    }
+
+    private static Command parseDeadlineInput(String input) {
+        try {
+            String taskDescription = input.replaceAll("/by.*", "").trim();
+            validateDescription(taskDescription, "deadline");
+            String deadline = input.split("/by")[1].trim();
+            return new DeadlineCommand(taskDescription, deadline);
+        } catch (ArrayIndexOutOfBoundsException e) {
+            throw new DukeException("The end time of a deadline cannot be empty.");
+        }
+    }
+
+    private static Command parseDeleteInput(String input) {
+        try {
+            validateDescription(input, "delete");
+            int taskNo = Integer.parseInt(input) - 1;
+            return new DeleteCommand(String.valueOf(taskNo));
+        } catch (NumberFormatException e) {
+            throw new DukeException("Enter a number for a delete action");
+        }
+    }
+
+    private static Command parseDoneInput(String input) {
+        try {
+            validateDescription(input, "done");
+            int taskNo = Integer.parseInt(input) - 1;
+            return new DoneCommand(String.valueOf(taskNo));
+        } catch (NumberFormatException e) {
+            throw new DukeException("Enter a number for a done action!");
```

Comment:
Would it make the error clearer to the user if you had different exception classes extending DukeException?

Reply:
haha yes